log/logger.go

```diff
@@ -155,7 +155,7 @@ func (l nullLogger) WithFields(newFields ...Field) Logger {
 		fields = append(fields, existingFields.ValueAt(i))
 	}
 	fields = append(fields, newFields...)
-	return nullLogger{Fields(fields)}
+	return &nullLogger{Fields(fields)}
```

Comment:
If we return a new logger here, don't we lose the ability to compare `NullLogger`s?

Reply:
Yeah good point, in that case I'll remove the fields from the `NullLogger` struct since it's not logging anything regardless.